### ISSUE TITLE
Patch Series

### ISSUE THREAD

**Issue by stephenfin** [getpatchwork/patchwork#9]
_From @stephenfin on July 27, 2015 15:9_
# Overview

A patch series is a collection of patches. In a patch series, each subsequent patch is usually dependent on its predecessor. This can be visualized like so:

```
- 0/5  # shallow threading
  - 1/3
  - 2/3
  - 3/3
```

Or this:

```
- 0/5  # deep threading (pre-Git 1.7.0 default behavior)
  - 1/3
    - 2/3
      - 3/3
```

It should be possible to link patches together using the `references:` header, defined in [RFC2822](https://www.ietf.org/rfc/rfc2822.txt). @dlespiau has [done some work here](https://github.com/dlespiau/patchwork/commit/3ff54b06c6be724c902faa7f1c6931a298d9c331) which we can use.

The end result of this would be as follows:
- A table displaying the dependencies/follow-ups to a patch
- An option to download an entire series
- An extension to the XML-RPC API to allow downloading of the series

We may also wish to add the following:
- A "series" view, similar to the patches view, that will display cover letter information
# Dependencies

This relies on #23 being implemented.
# Additional Work Items

An additional work item would be to write a script to create Jenkins jobs, reporting results using the status API provided by #12.
# Alternatives

TODO
# Useful resources

TODO

_Copied from original issue: stephenfin/patchwork#25_

**Comment by ruscur** [comment]
How would the API URLs work for this?  Would there be a series API that was separate to that of patches and contained references to each patch, and would each patch contain a reference to the series that it belongs to and/or a pointer to the next/previous patch in the series?

Also, maybe versions of a series could be added into this such that a patch series "object" always refers to the latest version of that series, or gets updated to point to other versions of the series.

**Comment by stephenfin** [comment]
@ruscur Sorry I never spotted this question earlier. I favor the latter of the approaches you described above. In my mind, patches are the pillar of patchwork and we shouldn't supplant these with a new "Series" object. I'd be in favor of adding a series reference and using this to sort things.

**Comment by stephenfin** [comment]
This is now merged and will form part of the upcoming 2.0 release
